### ISSUE TITLE
Cutom assets folder not resolved in current module scope

### ISSUE THREAD

**Issue by ogerardin** [fvarrui/JavaPackager#38]
I have a main POM with submodules, one of which is dedicated to building distribution packages and installers. So my structure is like:
```
pom.xml
module1
    pom.xml
...
dist
    pom.xml
```

The javapackager plugin is of course in the dist/pom.xml, but if I want to provide a custom icon I must put the assets folder at the main pom level:
```
pom.xml
assets
    mac
        xxx.icns
    ...
module1
    pom.xml
...
dist
    pom.xml
```

I suggest that customized assets should be resolved in the scope of the plugin's current module, i.e.:
```
pom.xml
module1
    pom.xml
...
dist
    assets
        mac
            xxx.icns
        ...
    pom.xml
```

**Comment by fvarrui** [comment]
Hi @ogerardin,
now it's using Maven execution dir as base to locate `assets` ... do you think it would be a good idea to use `${project.basedir}` property as the parent of `assets` directory?

**Comment by ogerardin** [comment]
> now it's using Maven execution dir as base to locate `assets` ... do you think it would be a good idea to use `${project.basedir}` property as the parent of `assets` directory?

Definitely :) The assets folder should be relative to the projet using the plugin.
If you want to keep compatibility you can search both places.
I was also thinking if the assets folder is a resource (i.e. in src/main/resources), it will be copied to the target dir and so you could also look in ${project.build.directory}

**Comment by fvarrui** [comment]
I think to use `${basedir}` as parent folder for `assets` and include a new property (maybe called `assetsDir`) to specify a different `assets` folder location.

I'm not sure if put `assets` in `src/main/resources` is a good idea, because maybe you don't want to distribute .icns or .ico files, which are only necessary to build installation artifacts.

**Comment by ogerardin** [comment]
Good idea to add a property, now you can only do it for icon!
You're right if it's a resource it will end up in the JAR which we probably don't want...

**Comment by fvarrui** [comment]
New property `assetsDir` added, so now it allows to customize assets folder, where icons and custom templates are stored. This property points to `${basedir}/assets` by default to maintain backward compatibility. Now it has to resolve assets in the scope of the plugin's current module.

Please, could you give a try to **1.0.3-SNAPSHOT**?

**Comment by ogerardin** [comment]
Yep it works :)

I replaced 
`<iconFile>${project.basedir}/assets/mac/X-Plane Manager.icns</iconFile>`
with
`<assetsDir>${project.basedir}/assets</assetsDir>`

And since it is now the default I was able to remove it completely!

**Comment by fvarrui** [comment]
When we definitely fixed all your issues, I’ll release v1.0.3 to Maven Central. Thanks!
